xbmc/interfaces/json-rpc/FileOperations.cpp

```diff
@@ -247,7 +251,7 @@ JSONRPC_STATUS CFileOperations::SetFileDetails(const std::string &method, ITrans
   CVideoLibrary::UpdateResumePoint(parameterObject, infos, videodatabase);
 
   videodatabase.GetFileInfo("", infos, fileId);
-  CJSONRPCUtils::NotifyItemUpdated(infos);
+  CJSONRPCUtils::NotifyItemUpdated(infos, artwork);
```

Comment:
Avoid the DB call by passing in a completely empty `artwork` here, which should put the artwork map back into the state the receiver expects - empty if artwork isn't updated, so the GUI item will keep the artwork it already has.

Reply:
Not sure that I can see that @rmrector `CGUIListItem::SetArt` simply replaces the art map, no empty artwork check.

Reply:
[The check happens later](https://github.com/xbmc/xbmc/blob/master/xbmc/FileItem.cpp#L1643), when the receiver of this message tells the original FileItem to update from the new FileItem in the message built inside of `CJSONRPCUtils::NotifyItemUpdated`.

An empty art map here restores this message to its original shape, which restores the behavior we want - `SetFileDetails` can't set artwork, so there is no reason to update artwork - the GUI ListItem can keep the artwork it has.

Reply:
Thanks for clarifying. I had concerns about SetFileDetails wanting to read art from DB when art isn't something that it can set, but couldn't quite it pin down.

Reply:
Done as suggested. Hopefully done right?